src/Storages/System/StorageSystemDataSkippingIndices.cpp

```diff
@@ -121,6 +124,14 @@ class DataSkippingIndicesSource : public ISource
                     // 'type' column
                     if (column_mask[src_index++])
                         res_columns[res_index++]->insert(index.type);
+                    // 'type_full' column
+                    if (column_mask[src_index++])
+                    {
+                        if (auto * expression = index.definition_ast->as<ASTIndexDeclaration>())
+                            res_columns[res_index++]->insert(queryToString(expression->type->clone()));
```

Comment:
Why clone()?

Reply:
It was first because of an incomplete definition of `ASTFunction`, and then the function `queryToString` didn't had a definition who accept raw pointers.

I modified your last changes with adding the prototype for `queryToString` if it's what we want